optaplanner-website-root/content/download/upgradeRecipe/upgradeRecipe8.adoc

```diff
@@ -1853,3 +1853,67 @@ After in `*.java`:
 ----
 
 For more information about these changes, refer to https://www.optaplanner.org/docs/optaplanner/latest/score-calculation/score-calculation.html#explainingTheScore[Explaining the score] in https://www.optaplanner.org/docs/[OptaPlanner documentation].
+
+[.upgrade-recipe-recommended.upgrade-recipe-public-api]
```

Comment:
This will definitely be `major`+`public-api` in the upgrade recipe for 9.0 where the old `@CustomShadowVariable` will be removed. But I'm not sure how to tag this here in the 8.0 recipe.

Reply:
Deprecations are "major", but not "public api" because the code still works.
"public api" is when the deprecated methods get removed - in a x.0.

So for here, just major is enough.